enso/experiment/rationalized.py

```diff
@@ -92,4 +95,211 @@ def predict(self, X, **kwargs):
     def cleanup(self):
         del self.model
 
-                                                                                                
+
+@Registry.register_experiment(ModeKeys.RATIONALIZED, requirements=[("Featurizer", "PlainTextFeaturizer")])
+class ReweightedGloveClassifier(ClassificationExperiment):
+
+    NLP = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(auto_resample=False, *args, **kwargs)
+        self.model = LogisticRegression()
+        self.p_rationale_given_word = {}
+        if self.NLP is None:
+            self.NLP = spacy.load('en_vectors_web_lg')
+   
+    def _compute_p_rationale(self, docs, rationale_docs):
+        word_counts = Counter([
+            token.text.lower() for doc in docs for token in doc
+        ])
+        rationale_word_counts = Counter([
+            token.text.lower() for doc in rationale_docs for token in doc
+        ])
+
+        # smoothing for unseen terms
+        base_freq = np.sqrt(1. / sum(word_counts.values()))
+        self.p_rationale_given_word = defaultdict(lambda: base_freq)
+
+        for word, count in rationale_word_counts.items():
+            # there can be slight difference in tokenization -- so we set 
+            # word counts to be the rationale word count when we encounter this
+            self.p_rationale_given_word[word] = np.sqrt(count / word_counts.get(word, count))
+
+        return self.p_rationale_given_word
+
+    def _featurize(self, doc):
+        doc_vect = np.mean([
+            token.vector * self.p_rationale_given_word[token.text.lower()]
+            for token in doc
+        ], axis=0)
+        normed_doc_vect = doc_vect / np.linalg.norm(doc_vect)
```

Comment:
Have you tried it without this normalisation?

Reply:
Nope but I should for sure